### PR TITLE
Feature | Agency repository

### PR DESCRIPTION
**Pull request by aalban3**
## Description
https://app.clickup.com/t/86dub4k5j

Aside from ticket specific tasks I added:
- Required packages for Unit testing
- A class that serves as a mock agency factory
- A class to help create an in-memory DB for testing

## Task List

- [x] Unit Tests
- [ ] Integration Tests
- [ ] Design Updates

### CODE REVIEW

File: Dashdoc.API/Dashdoc.API.Data/EntityConfigurations/AgencyConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public class AgencyConfiguration: IEntityTypeConfiguration<Agency>
+{
+    public void Configure(EntityTypeBuilder<Agency> builder)
+    {
+        builder.HasKey(e => e.Id);
+        
+        builder.Property(e => e.Name).IsRequired();
+        builder.Property(e => e.Email).IsRequired();
+
+        builder.HasMany(e => e.Providers);
```
Comment (beemak789) [inline Dashdoc.API/Dashdoc.API.Data/EntityConfigurations/AgencyConfiguration.cs:16]: Why is this added into the agency configuration?

File: Dashdoc.API/Dashdoc.API.sln [hunk 2]
```diff
@@ -16,6 +16,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dashdoc.API.Data.Tests", "D
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dashdoc.API.Data", "Dashdoc.API.Data\Dashdoc.API.Data.csproj", "{A9220C78-753C-4700-AFF2-472D04B4AFBD}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dashdoc.Tests.Common", "Dashdoc.Tests.Common\Dashdoc.Tests.Common.csproj", "{6C0905B0-E2A9-4E78-9BE9-C50E0CAF4DE2}"
```
Comment (beemak789) [inline Dashdoc.API/Dashdoc.API.sln:19]: What are these? should these be in env variables?

Comment (aalban3) [reply Dashdoc.API/Dashdoc.API.sln:19]: This is autogenerated by dotnet

File: Dashdoc.API/Dashdoc.API.Data/EntityConfigurations/AgencyConfiguration.cs (hunk 1 above)
Comment (aalban3) [reply Dashdoc.API/Dashdoc.API.Data/EntityConfigurations/AgencyConfiguration.cs:16]: That's the relation it has to providers

**Review by beemak789 — APPROVED** [review]
